src/Microsoft.IdentityModel.Protocol.Extensions/OpenIdConnectMessage.cs

```diff
@@ -137,12 +139,30 @@ public string CreateAuthenticationRequestUrl()
         /// <returns>The uri to use for a redirect.</returns>
         public string CreateLogoutRequestUrl()
         {
+            EnsureTelemetryValues();
             OpenIdConnectMessage openIdConnectMessage = Clone();
             openIdConnectMessage.RequestType = OpenIdConnectRequestType.LogoutRequest;
             return openIdConnectMessage.BuildRedirectUrl();
         }
 
         /// <summary>
+        /// Adds telemetry values to the message parameters.
+        /// </summary>
+        private void EnsureTelemetryValues()
+        {
+            if (DisableTelemetryParameters)
```

Comment:
Yes, if you were guaranteed that Clone was always called before EnsureTelemetryValues. My concern is that, in some future iteration of the code, someone changes the flow to not call Clone first, and then they expect that setting DisableTelemetryValues to true, and then to false, would remove them again. I can see your point though; if you think we shouldn't set up the code to ever remove the parameters, I'm fine with changing it.